### ISSUE TITLE
Changing MAX_CAMERAS

### ISSUE THREAD

**Issue by akihikoy** [opencv/opencv#21033]
In [videoio/src/cap_v4l.cpp](https://github.com/opencv/opencv/blob/17234f82d025e3bbfbf611089637e5aa2038e7b8/modules/videoio/src/cap_v4l.cpp), `MAX_CAMERAS` is defined as 8, so opening a camera `/dev/video8` fails with an error like `VIDEOIO ERROR: V4L: index 8 is not correct!`. 

https://github.com/opencv/opencv/blob/98b6ce353cc9342bee472e594eaf4b23136b0430/modules/videoio/src/cap_v4l.cpp#L271

This value was set because it is almost impossible to open 8 cameras simultaneously due to the USB bandwidth. However, nowadays, the situation is different. Some cameras (e.g. Intel RealSense) uses multiple `/dev/video*` entries (e.g. [Four /dev/video entries but just one camera](https://askubuntu.com/questions/1123601/four-dev-video-entries-but-just-one-camera)). Thus, the camera index easily exceeds 8. In my particular case, 3 cameras are connected (2xVGA, 1xRealSense) where there are 10 /dev/video* entries!
```
$ v4l2-ctl --list-devices
VGA Camera: VGA Camera (usb-0000:00:14.0-10.3):
        /dev/video8
        /dev/video9

Intel(R) RealSense(TM) Depth Ca (usb-0000:00:14.0-2.4):
        /dev/video0
        /dev/video1
        /dev/video2
        /dev/video3
        /dev/video4
        /dev/video5

VGA Camera: VGA Camera (usb-0000:00:14.0-3.4):
        /dev/video6
        /dev/video7
```
Note that I can simultaneously use those 3 cameras.

Thus, I suggest to increase `MAX_CAMERAS` value.

-------------
(Edited)
This is actually not a critical issue. `cv::VideoCapture cap(8)` fails, but we can open the camera with `cv::VideoCapture cap("/dev/video8")`.

**Comment by alalek** [comment]
> `cv::VideoCapture cap("8")`

`cv::VideoCapture cap(8);` should not fail too as `MAX_CAMERAS` is used with `index < 0` only:

https://github.com/opencv/opencv/blob/98b6ce353cc9342bee472e594eaf4b23136b0430/modules/videoio/src/cap_v4l.cpp#L852-L854

**Comment by akihikoy** [comment]
I'm using OpenCV 3.2.0 (binary version of Ubuntu 16.04). I double-checked that I got error for `cv::VideoCapture cap("8")` but worked with `cv::VideoCapture cap("/dev/video8")`.
The error `VIDEOIO ERROR: V4L: index 8 is not correct!` seems to be printed here:

https://github.com/opencv/opencv/blob/70bbf17b133496bd7d54d034b0f94bd869e0e810/modules/videoio/src/cap_v4l.cpp#L792-L797

`indexList` looks depending on `MAX_CAMERAS` (in this version).

However I could not find the corresponding part in newer branches (3.4, 4.x), so the issue might be already fixed (not by changing `MAX_CAMERAS`)? If so, changing `MAX_CAMERAS` is my misunderstanding. Please close this topic.

**Comment by alalek** [comment]
The right usage is "8" => 8 without quotes.

**Comment by akihikoy** [comment]
Yes, actually I was writing `cv::VideoCapture cap(8)` in my code (and, got error with it). I edited above my messages.

**Comment by alalek** [comment]
Problem is already resolved for 3.4.2 release here: 83d915eca1f4f3e1cb28b2d7444890d911e7f9a9
We don't support old releases.

You need to upgrade.
